beacon-chain/blockchain/core.go

```diff
@@ -222,18 +222,22 @@ func (b *BeaconChain) verifyBlockTimeStamp(block *types.Block) (bool, error) {
 // computeNewActiveState for every newly processed beacon block.
 func (b *BeaconChain) computeNewActiveState(attestations []*pb.AttestationRecord, activeState *types.ActiveState, blockVoteCache map[[32]byte]*types.VoteCache, blockHash [32]byte) (*types.ActiveState, error) {
```

Comment:
You're on the right track. I would actually move this function into `types/state.go`. Something along the lines of:

```
func (a *ActiveState) computeNewActiveState(attestations, blockVoteCache, blockHash) *ActiveState {
    ...
}
```

Inside this function, you have easy access to all of the fields that you need, and you won't have to use accessor methods like `ReplaceBlockHashes`. Also, I see that you manually copied the elements in `activeState.RecentBlockHashes()` into a new slice. See if you can do it more succinctly using `copy`.